pkg/cloudprovider/cloudprovider.go

```diff
@@ -232,6 +233,41 @@ func (c *CloudProvider) waitUntilLaunched(ctx context.Context, nodeClaim *karpv1
 	}
 }
 
+// handleNodeClaimCreationError handles common error processing for both standalone and async node claim creation failures
+func (c *CloudProvider) handleNodeClaimCreationError(ctx context.Context, err error, promise *instance.VirtualMachinePromise, nodeClaim *karpv1.NodeClaim, removeNodeClaim bool) error {
+	c.recorder.Publish(cloudproviderevents.NodeClaimFailedToRegister(nodeClaim, err))
```

Comment:
This is net-new, right?

Are there other events we should be emitting?

Do we have an issue tracking them (or should we?)